rfcs/text/0006_management_section_service.md

```diff
@@ -0,0 +1,284 @@
+kibanaSection.register('mypluginlink', {
+  name: 'mypluginlink',
+  order: 10,
+  display: 'My sublink',
+  url: `#/management/myplugin`,
+});
+ 
+// use ui/routes to render component
+import routes from 'ui/routes';
+ 
+const renderReact = (elem) => {
+  render(<MyApp />, elem);
+};
+ 
+routes.when('management/myplugin', {
+  controller($scope, $http, kbnUrl) {
+    $scope.$on('$destroy', () => {
+      const elem = document.getElementById('usersReactRoot');
+      if (elem) unmountComponentAtNode(elem);
+    });
+    $scope.$$postDigest(() => {
+      const elem = document.getElementById('usersReactRoot');
+      const changeUrl = (url) => {
+        kbnUrl.change(url);
+        $scope.$apply();
+      };
+      renderReact(elem, $http, changeUrl);
+    });
+  },
+});
+```
+Current public contracts owned by the legacy service:
+```js
+// ui/management/index
+interface API {
+  PAGE_TITLE_COMPONENT: string; // actually related to advanced settings?
+  PAGE_SUBTITLE_COMPONENT: string; // actually related to advanced settings?
+  PAGE_FOOTER_COMPONENT: string; // actually related to advanced settings?
+  SidebarNav: React.SFC<any>;
+  registerSettingsComponent: (
+    id: string,
+    component: string | React.SFC<any>,
+    allowOverride: boolean
+  ) => void;
+  management: new ManagementSection();
+  MANAGEMENT_BREADCRUMB: {
+    text: string;
+    href: string;
+  };
+}
+
+// ui/management/section
+class ManagementSection {
+  get visibleItems,
+  addListener: (fn: function) => void,
+  register: (id: string, options: Options) => ManagementSection | Error,
+  deregister: (id: string) => void,
+  hasItem: (id: string) => boolean,
+  getSection: (id: string) => ManagementSection,
+  hide: () => void,
+  show: () => void,
+  disable: () => void,
+  enable: () => void,
+}
+ 
+interface Options {
+  order: number | null;
+  display: string | null; // defaults to id
+  url: string | null; // defaults to ''
+  visible: boolean | null; // defaults to true
+  disabled: boolean | null; // defaults to false
+  tooltip: string | null; // defaults to ''
+  icon: string | null; // defaults to ''
+}
+```
+
+# Notes
+
+- The hide/show/disable/enable options were dropped with the assumption that we will be working with uiCapabilities to determine this instead... so people shouldn't need to manage it manually as they can look up a pre-filtered list of sections.
+- This was updated to add flexibility for custom (non-EUI) icons as outlined in #32661. Much like the Core Application Service, you either choose an EUI icon, or provide a URL to an icon.
+
+# Drawbacks
+
+- This removes the ability to infinitely nest sections within each other by making a distinction between a section header and a nav link.
+  - So far we didn't seem to be using this feature anyway, but would like feedback on any use cases for it.
+
+# Unresolved questions
+
+- Do individual links within a section need the ability to specify icons? Currently only the section headers have icons.
```

Comment:
IMO we can leave out link icons and description unless we have concrete plans to use them.